src/app/cli/src/coda_worker.ml

```diff
@@ -240,19 +243,29 @@ module T = struct
              ~snark_pool_disk_location:(conf_dir ^/ "snark_pool")
              ~time_controller ~receipt_chain_database
              ~snark_work_fee:(Currency.Fee.of_int 0)
-             ?propose_keypair:Config.propose_keypair () ~banlist)
+             ?propose_keypair:Config.propose_keypair () ~banlist ~monitor)
       in
-      Option.iter snark_worker_config ~f:(fun config ->
-          let run_snark_worker = `With_public_key config.public_key in
-          Run.setup_local_server ~client_port:config.port ~coda ~log () ;
-          Run.run_snark_worker ~log ~client_port:config.port run_snark_worker
-      ) ;
-      let coda_peers () = return (Main.peers coda) in
-      let coda_start () = return (Main.start coda) in
+      Run.handle_shutdown ~monitor ~frontier_file ~log coda ;
+      let%bind () =
+        with_monitor
+        @@ fun () ->
+        return
+        @@ Option.iter snark_worker_config ~f:(fun config ->
+               let run_snark_worker = `With_public_key config.public_key in
+               Run.setup_local_server ~client_port:config.port ~coda ~log () ;
+               Run.run_snark_worker ~log ~client_port:config.port
+                 run_snark_worker )
+      in
+      let coda_peers () = with_monitor @@ fun () -> return (Main.peers coda) in
```

Comment:
instead of putting `with_monitor` here, I'd put it around the functions put into the record below. Like

```ocaml
return
  { coda_peers= with_monitor coda_peers
  ...
```

I think that way it will be harder to forget to return something that doesn't have the monitor.